src/models/TechRecordsDAO.ts

```diff
@@ -132,6 +129,24 @@ class TechRecordsDAO {
       },
       ReturnValues: "ALL_NEW"
     };
+    if (techRecord.primaryVrm) {
+      query.UpdateExpression += ", primaryVrm = :primaryVrm";
+      Object.assign(query.ExpressionAttributeValues, {
+        ":primaryVrm": techRecord.primaryVrm
+      });
+    }
+    if (techRecord.secondaryVrms && techRecord.secondaryVrms.length) {
+      query.UpdateExpression += ", secondaryVrms = :secondaryVrms";
+      Object.assign(query.ExpressionAttributeValues, {
+        ":secondaryVrms": techRecord.secondaryVrms
+      });
+    }
+    if (techRecord.trailerId) {
+      query.UpdateExpression += ", trailerId = :trailerId";
+      Object.assign(query.ExpressionAttributeValues, {
+        ":trailerId": techRecord.trailerId
+      });
+    }
```

Comment:
if we are updating the complete `techRecord` on line 123 then why are why appending separate query parameters.

Reply:
UpdateExpression: "set techRecord = :techRecord" means "update just the techRecord array". If you want to update the primaryVrm or trailerId or other attributes on the vehicle object you have to specify it.

I'm appending separate query parameters, because for example if you're updating a PSV you won't have trailerId on it, it will be undefined and in ExpressionAttributeValues if you try and set trailerId: undefined it will fail because it has to be defined.